src/common/validators.ts

```diff
@@ -241,18 +238,10 @@ const checkObjectNameAsyncValidation = (
       }
 
       if (isMultiRegionEnabled) {
-        if (isExistingRegionalClusterName) {
-          return false;
-        }
-      } else {
-        const search = `name = ${sqlString(value)}`;
-        const { data } = await clusterService.searchClusters(search, 1);
-        // Normally, we get 0 or 1 items, 1 meaning a cluster of that name already exists.
-        // But dumb mockserver ignores `search` and `size`, always returns full static list;
-        // checking the returned name(s) allows this validation to work in ?env=mockdata UI.
-        return !data?.items?.some((cluster) => cluster.name === value);
+        return !isExistingRegionalClusterName;
       }
-      return true;
+
+      return isExistingRegionalClusterName !== undefined ? !isExistingRegionalClusterName : true;
```

Comment:
The logic is changing significantly here. When there's no multi region and there's no regional cluster name verification. That's the OSD scenario. In that case the validator will always return `true` now.

That's causing the async validation to be skipped in the OSD wizard. There's no API check anymore if the cluster name is already used now.

Can you double check please?

Reply:
This code was the original issue.  We were calling the API directly instead of using the react query.  The react query will be triggered externally to this and the values will be updated.  Are you seeing an actual bug now in the OSD path?  I believe I checked all paths but I will check it again.